### ISSUE TITLE
revisions RootQuery returns a 500 when id is included in query

### ISSUE THREAD

**Issue by justinwhall** [wp-graphql/wp-graphql#907]
👋 
**WPGraphql version**: 0.3.4
**Other Plugins of significance:** WPGraphQL JWT Authentication

If I run the the following query as an _unathenticated_ user... 
```
query GET_REVISIONS($postId: Int){
  revisions{
    nodes {
      title
    }
  }
postBy(postId: $postId) {
    slug
    id
    revisions {
      nodes {
        id
        title
      }
    }
  }
}
```
I get a response that that look like the following:
```
{
  "error": {
    "data": {
      "revisions": {
        "nodes": [
          {
            "title": null
          },
          {
            "title": null
          },
          {
            "title": null
          },
          {
            "title": null
          },
          {
            "title": null
          },
          {
            "title": null
          },
          {
            "title": null
          },
          {
            "title": null
          },
          {
            "title": null
          },
          {
            "title": null
          }
        ]
      },
      "postBy": {
        "slug": "new-expert-services-offerings-for-senders-of-every-size-and-budget",
        "id": "cG9zdDozOTcxNg==",
        "revisions": {
          "nodes": []
        }
      }
    }
  }
}
```
I'd expect the error nodes  array to be empty like `postBy` object ie: 

```
error: { data: { revisions: { nodes : [] } } }
```

If I then add `id` to the `revisions` query, I get a "internal server error" response:

**Query**
```
query GET_REVISIONS($postId: Int){
  revisions{
    nodes {
      title
      id <--- I'm new here.
    }
  }
postBy(postId: $postId) {
    slug
    id
    revisions {
      nodes {
        id
        title
      }
    }
  }
}
```
**Response**
```
{
  "error": {
    "errors": [
      {
        "message": "Internal server error",
        "category": "internal",
        "locations": [
          {
            "line": 5,
            "column": 7
          }
        ],
        "path": [
          "revisions",
          "nodes",
          0,
          "id"
        ]
      },
     // ^^ This repeats for each revision of course.
    "data": {
      "revisions": {
        "nodes": [
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null,
          null
        ]
      },
      "postBy": {
        "slug": "new-expert-services-offerings-for-senders-of-every-size-and-budget",
        "id": "cG9zdDozOTcxNg==",
        "revisions": {
          "nodes": []
        }
      }
    }
  }
}
```
I went to investigate a possible patch, but unlike other errors, I'm not seeing anything logged to debug.log.

Happy to provide an endpoint for you to test if you want to contact me offline.

**Comment by CodeProKid** [comment]
I think you're getting `NULL` returned by each of the nodes because the model layer is catching these with an unauthenticated user. We should probably also catch this before the query runs, and just return an empty array.

For the internal server error, I believe it's because the server is trying to get a field off of a null value, and just isn't handling it very gracefully. We can probably do better with error reporting for stuff like this.

**Comment by jasonbahl** [comment]
ah, ya. @CodeProKid you're correct. We need to address this much like we do stati (future, etc) that a user cannot query for: https://github.com/wp-graphql/wp-graphql/blob/develop/src/Data/Connection/PostObjectConnectionResolver.php#L419-L433

We should do the same with `post_type(s)` 🤔

**Comment by jasonbahl** [comment]
Actually, looks like we _should_ be avoiding querying revisions if users don't have proper caps:

https://github.com/wp-graphql/wp-graphql/blob/develop/src/Data/Connection/PostObjectConnectionResolver.php#L84-L96

Perhaps something here in this logic is messed up 🤔

**Comment by jasonbahl** [comment]
Here's the logic in the model layer re: revisions

https://github.com/wp-graphql/wp-graphql/blob/develop/src/Model/Post.php#L201-L203

https://github.com/wp-graphql/wp-graphql/blob/develop/src/Model/Post.php#L217-L230

So, something in my logic at the resolver level is letting it pass through, but then the model layer is blocking it. 🤔

**Comment by jasonbahl** [comment]
ah yes, that's because we need some context about the parent to decide at the model layer. . .we can't _fully_ determine at the resolver level before we query if the user has permission or not.

🤔

**Comment by TylerBarnes** [comment]
From other discussions it sounds like we're going to re-do how revisions are handled and potentially make a breaking change to remove the revisions field. Should we close this issue since it will probably be addressed but in a different way?

**Comment by TylerBarnes** [comment]
This is the issue https://github.com/wp-graphql/wp-graphql/issues/974

**Comment by jasonbahl** [comment]
This is addressed in the v0.6.0 release. 

See it in action, querying for revisions as an unauthenticated user: 

![Screen Shot 2020-01-14 at 1 52 12 PM](https://user-images.githubusercontent.com/1260765/72381710-17981c00-36d5-11ea-95bc-eeb619c59863.png)
